File: BringMeHome/BringMeHome.Services/Database/Breed.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BringMeHome.Services.Database
{
    public class Breed
    {
        [Key]
        public int BreedID { get; set; }

        public int SpeciesID { get; set; }

        [Required, StringLength(100)]
        public string BreedName { get; set; }

        [StringLength(500)]
        public string Description { get; set; }

        [StringLength(50)]
        public string SizeCategory { get; set; }

        [StringLength(500)]
        public string TemperamentNotes { get; set; }

        [StringLength(500)]
        public string SpecialNeeds { get; set; }

        [StringLength(500)]
        public string CommonHealthIssues { get; set; }

        // Navigation properties
        [ForeignKey("SpeciesID")]
        public virtual Species Species { get; set; }

        public virtual ICollection<Animal> Animals { get; set; }
    }
}

File: bringmehomeUI/bringmehome_admin/lib/models/breed.dart
class Breed {
  int breedID;
  String breedName;
  int? speciesID;

  Breed({
    required this.breedID,required this.breedName,this.speciesID,
  });

  factory Breed.fromJson(Map<String, dynamic> json) {
    return Breed(
      breedID: json['breedID'] as int,
      breedName: json['breedName'] as String,
      speciesID: json['speciesID'] != null ? json['speciesID'] as int : null,
    );
  }
}